lsp-positions/src/lib.rs

```diff
@@ -44,6 +44,7 @@ fn utf16_len(string: &str) -> usize {
 /// All of the position information that we have about a character in a source file
 #[repr(C)]
 #[derive(Clone, Debug, Default, Eq, Hash, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
That's very neat! Didn't know you could make `derive` conditional like that.

Reply:
it is! another cool use-case is in combination with the builder/[typed-builder](https://docs.rs/typed-builder/latest/typed_builder/) crates, to do something like `#[cfg_attr(feature = "enable_builders", derive(TypedBuilder))]`, to conditionally allow the builder pattern.